src/xwayland.c

```diff
@@ -478,6 +494,12 @@ handle_set_decorations(struct wl_listener *listener, void *data)
 	view_set_decorations(view, want_deco(xwayland_view->xwayland_surface));
 }
 
+static void
+handle_set_window_type(struct wl_listener *listener, void *data)
```

Comment:
What's this one for?

Reply:
Should we connect the signal in readiness?

Reply:
It was originally connected as part of the example but I left it in there to make it easier to actually use the feature.
I can remove the signal handler though if preferred, technically it isn't necessary for `xwayland_surface_contains_window_type()`.